### PR TITLE
Upgrade to Sylius 1.10 and Symfony >5.2

### PR DESCRIPTION
**Pull request by Roshyo**
Provide support for Sylius 1.10 and Symfony 5.2 or later

### CODE REVIEW

**Comment by Zales0123** [comment]
Hello @Roshyo! Even though we're not supporting this plugin actively nowadays, this PR is definitely something we would like to have :) I've enabled proper builds and made some changes in dependencies, but it requires a little bit of work to be mergeable. Do you think you would be able to fix it in the nearest future, or should I take it over?

Cheers 🖖

**Comment by Roshyo** [comment]
Well, since I will be away for the next 3 weeks or so. I would love it if you could take over on that point.

Or it will have to wait until I get back.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -18,16 +18,19 @@ jobs:
...
+                php: [7.4, 8.0]
                 symfony: [^4.4, ^5.2]
-                sylius: [~1.9.0]
+                sylius: [1.9.*, 1.10.*]
                 node: [10.x]
                 mysql: [5.7, 8.0]
 
                 exclude:
                     - # PHP 7.3 does not support "caching_sha2_password" authentication plugin which is a default one in MySQL 8.0
                         php: 7.3
```
Comment (GSadee) [inline .github/workflows/build.yml:29]: This exclusion is not needed anymore

**Review by Zales0123 — APPROVED** [review]

File: composer.json [hunk 2]
```diff
@@ -19,29 +19,31 @@
...
-        "php": "^7.3",
+        "php": "^7.4 || ^8.0",
 
-        "sylius/sylius": "~1.9.0",
+        "sylius/sylius": "1.9.*",
         "friendsofsymfony/jsrouting-bundle": "^2.2",
-        "friendsofsymfony/oauth-server-bundle": ">2.0.0-alpha.0 ^2.0@dev"
+        "symfony/flex": "^1.10",
+        "symfony/asset": "^4.4 || ^5.2",
+        "friendsofsymfony/oauth-server-bundle": "^1.6 || >2.0.0-alpha.0 ^2.0@dev"
```
Comment (pamil) [inline composer.json:28]: Might be considered as dev-dependency only (Sylius installations will have it defined in the root package if needed).

File: composer.json [hunk 3]
```diff
@@ -19,29 +19,31 @@
...
     ],
     "require": {
-        "php": "^7.3",
+        "php": "^7.4 || ^8.0",
 
-        "sylius/sylius": "~1.9.0",
+        "sylius/sylius": "1.9.*",
         "friendsofsymfony/jsrouting-bundle": "^2.2",
-        "friendsofsymfony/oauth-server-bundle": ">2.0.0-alpha.0 ^2.0@dev"
+        "symfony/flex": "^1.10",
```
Comment (pamil) [inline composer.json:26]: This is not needed for a correct installation.

File: composer.json [hunk 4]
```diff
@@ -19,29 +19,31 @@
...
+        "symfony/asset": "^4.4 || ^5.2",
```
Comment (pamil) [inline composer.json:27]: This shouldn't be needed.

File: composer.json [hunk 5]
```diff
@@ -69,6 +71,10 @@
     "extra": {
         "branch-alias": {
             "dev-master": "1.1-dev"
+        },
+        "symfony": {
```
Comment (pamil) [inline composer.json:75]: This config might be deleted.

**Review by pamil — APPROVED** [review]

**Comment by pamil** [comment]
Thank you, Stephane! :tada:

**Comment by aleho** [comment]
The release notes state "Upgrade to Sylius 1.10" but this PR did not bump the allowed Sylius version from 1.9.

This resluts in `sylius/admin-order-creation-plugin v0.12.0 requires sylius/sylius 1.9.* -> found sylius/sylius[v1.9.0-RC.1, ..., v1.9.5] but it conflicts with your root composer.json require (1.10.*).`.

Has this PR not been tested with Sylius 1.10 or is my composer.json wrong?

**Comment by pamil** [comment]
@aleho good catch, we've totally missed that!

**Pull request merged**